libcontainer/env.go

```diff
@@ -0,0 +1,56 @@
+package libcontainer
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+)
+
+// prepareEnv checks supplied environment variables for validity, removes
+// duplicates (leaving the last value only), and sets PATH from env, if found.
+// Returns the deduplicated environment, and a flag telling if HOME is found.
+func prepareEnv(env []string) ([]string, bool, error) {
+	// Clear the current environment (better be safe than sorry).
+	os.Clearenv()
+
+	if env == nil {
+		return nil, false, nil
+	}
+	// Deduplication code based on dedupEnv from Go 1.22 os/exec.
+
+	// Construct the output in reverse order, to preserve the
+	// last occurrence of each key.
+	out := make([]string, 0, len(env))
+	saw := make(map[string]bool, len(env))
+	for n := len(env); n > 0; n-- {
+		kv := env[n-1]
```

Comment:
nit: It seems simpler to use n = len(env) -1, to keep the structure of loops we used for a lot of years already (but you need to enter on the n=o case too)

Reply:
> // Deduplication code based on dedupEnv from Go 1.22 os/exec.

As per the above comment, I stole this code from Go stdlib, and I don't want to change it from the original more than necessary.

Reply:
Ohh, I didn't know that. Makes sense, thanks :)